Source/Plugins/Core/com.equella.core/js/tsrc/modules/FileUploaderModule.ts

```diff
@@ -0,0 +1,403 @@
+  axiosSourceMap.set(uploadingFile.localId, source);
+  const token = source.token;
+
+  const formData = new FormData();
+  formData.append("file", fileEntry);
+  return Axios.post<UpdateEntry | UploadFailed>(path, formData, {
+    cancelToken: token,
+    onUploadProgress: (progressEvent: ProgressEvent) => {
+      updateUploadProgress({
+        ...uploadingFile,
+        uploadPercentage: Math.floor(
+          (progressEvent.loaded / fileEntry.size) * 100
+        ),
+      });
+    },
+  }).then(({ data }) => data);
+};
+
+/**
+ * Send a POST request to remove an uploaded file.
+ * @param path The request URL
+ * @param id The file ID which must be the server generated ID
+ */
+export const deleteUpload = (
+  path: string,
+  id: string
+): Promise<RemoveEntries> => {
+  const deleteData: DeleteUpload = { command: "delete", id: id };
+  return Axios.post<RemoveEntries>(path, deleteData).then(({ data }) => data);
+};
+
+/**
+ * Cancel an upload request.
+ * Returns a resolved promise for successful cancel or a rejected promise otherwise.
+ * @param id The file ID which must be the client generated ID
+ */
+export const cancelUpload = (id: string) => {
+  const axiosSource = getAxiosSource(id);
+  if (axiosSource) {
+    axiosSource.cancel();
+    axiosSourceMap.delete(id);
+  }
+};
+
+/**
+ * Update whether to show the attachment duplicate warning. Because the warning message UI
+ * is created on server side through 'ftl' template, we have to use jQuery to update the DOM.
+ *
+ * @param ctrlId The root ID of an Attachment Wizard Control
+ * @param display True to show the warning
+ */
+export const updateDuplicateMessage = (ctrlId: string, display: boolean) => {
+  const duplicateMessageDiv = document.querySelector(
+    `#${ctrlId}_attachment_duplicate_warning`
+  );
+  if (duplicateMessageDiv) {
+    if (display) {
+      duplicateMessageDiv.setAttribute("style", "display: inline");
+    } else {
+      duplicateMessageDiv.setAttribute("style", "display: none");
+    }
+  }
+};
+
+/**
+ * Update the error text of an Attachment Wizard Control. An example of calling this
+ * function is when the number of uploaded/uploading files is more than the maximum
+ * number of attachments. Similar to 'updateDuplicateMessage', jQuery is used to help
+ * update DOM.
+ *
+ * @param ctrlId The root ID of an Attachment Wizard Control
+ * @param text The text describing the error
+ */
+export const updateCtrlErrorText = (ctrlId: string, text: string) => {
+  const contElem = document.querySelector(`DIV#${ctrlId} > DIV.control`);
+  if (contElem) {
+    if (text === "") {
+      contElem.classList.remove("ctrlinvalid");
+    } else {
+      contElem.classList.add("ctrlinvalid");
+    }
+
+    const invalidMessage = contElem.querySelector("P.ctrlinvalidmessage");
+    if (invalidMessage) {
+      invalidMessage.textContent = text;
+    }
+  }
+};
+
+/**
+ * Build a text for the error of exceeding the maximum number of attachments
```

Comment:
```suggestion
 * Build a string for the error of exceeding the maximum number of attachments
```

But it'd be worth explaining the type of input strings it is expecting. Because I understand it, it's expecting a string/`format` like: `There were {0} duplicate files`.